### PR TITLE
[DOX] Corrected and expanded the (dependent) string set documentation

### PR DESCRIPTION
**Pull request by smehringer**
- Removed the deprecation of the *ById() functions
- Added a detailed description about the behaviour of the Dependent StringSet

this PR is related to #1794

### CODE REVIEW

**Comment by smehringer** [comment]
@h-2 I notice that this reverts a former issue #1289 


The documentation will now look like the following:

dox_dss

**Comment by h-2** [comment]
Does this also include documentation for the different specializations? 

Also you wanted to change `[n+1]` to undefined behaviour and `getValueById(DELETED_ELEMENT)` to empty element...

File: include/seqan/sequence/string_set_dependent_tight.h [hunk 1]
```diff
@@ -63,8 +63,30 @@ namespace seqan {
  * @tparam TString The type of the string to store in the string set.
  * @tparam TSpec   Tag for further specializing the string set.
  *
- * The class is not usable itself, only its subclasses @link TightDependentStringSet @endlink and
+ * Important: The class is not usable itself, only its subclasses @link TightDependentStringSet @endlink and
```
Comment (rrahn) [inline include/seqan/sequence/string_set_dependent_tight.h]: Please rather write something: ```This is an abstract class. Use one of the following specializations of an dependent string set: ... .

**Comment by smehringer** [comment]
@h-2 
not yet. I forgot to ask whether I should include the documentation for the different specializations on the same page (Dependent StringSet) or if I should document directly on each of the specialized sets?

I will add another pull request for the empty element issue since it's not related to the manual.

**Comment by h-2** [comment]
>  I forgot to ask whether I should include the documentation for the different specializations on the same page (Dependent StringSet) or if I should document directly on each of the specialized sets?

Add it to the specialization pages, but please include them in this PR.

> I will add another pull request for the empty element issue since it's not related to the manual.

But the documentation should also reflect this.

**Comment by smehringer** [comment]
> >I forgot to ask whether I should include the documentation for the different specializations on the same page (Dependent StringSet) or if I should document directly on each of the specialized sets?

> Add it to the specialization pages, but please include them in this PR.

I think my last commit (3e2e62d) did this?

> > I will add another pull request for the empty element issue since it's not related to the manual.

> But the documentation should also reflect this.

As we discussed last week, there will be no alteration of the behaviour of the different string set specializations (but I included in my last commit a documentation of how the string sets behave on semantic errors like empty element.)

**Comment by h-2** [comment]
AFAICT the getValueById for Tight dss is in O(log(n)) and not amortized constant. Can you double-check these values please?

**Pull request merged**